Yoast/yoast-acf-analysis
# Tests: use PSR-4

## Summary

This PR can be summarized in the following changelog entry:

* Tests: use PSR-4

## Relevant technical choices:

### Tests: move mock acf class to Doubles directory

### Tests: move unit tests to tests\Unit subdirectory

### Tests: rename some mock classes

... to comply with the YoastCS naming conventions.

These weren't being flagged so far as the tests are using a non-conventional directory layout.

### Tests: use PSR-4

* Ensure all test class names reflect the name of the class under test.
* Ensure all test class file names match the name of the test class.
* Change the Composer `autoload-dev` directive to use PSR-4 autoloading.
* Update the `phpunit.xml.dist` file to match.
* Update the `.phpcs.xml.dist` file to match.

## Test instructions

This PR can be tested by following these steps:

* _N/A_ If the tests pass, we're good.
